src/commands/faucet.ts

```diff
@@ -30,21 +31,33 @@ const FaucetModule = (cosmosClient: CosmosClient): Command => {
       return;
     }
 
+    const discordId = interaction.user.id;
     const address = interaction.options.getString('address', true);
 
     console.debug(`Token request to address ${address}`);
-
-    const rateLimited = rateLimiter.get(address);
-    if (rateLimited) {
+    const lastDrop = await Drop.findOne({ where: { discordId: discordId }, order: [['createdAt', 'DESC']] });
+    if (lastDrop
+      && (lastDrop.get('createdAt') as Date) > new Date(Date.now() - Config.faucet.cooldownInDay * 24 * 60 * 60 * 1000)
+    ) {
+      const txHash = lastDrop.get('txHash');
       await interaction.editReply(
-        `:negative_squared_cross_mark: Token already sent to address: \`${address}\`, Please come back again in a day`,
+        `:negative_squared_cross_mark: Each user can only claim once, txhash: [${txHash}](${Config.faucet.restUrl}/cosmos/tx/v1beta1/txs/${txHash}).`,
       );
       return;
     }
 
     try {
       const txHash = await cosmosClient.distribute(address);
-      rateLimiter.limit(address);
+      await Drop.create({
+        discordId: discordId,
+        address: address,
+        amount: Config.faucet.amount,
+        txHash: txHash,
+      });
```

Comment:
Would we have race conditions here?

Reply:
Run `Drop.create` first, then try catch `cosmosClient.distribute`,  revert the drop if distribute  fail
Also we need `sequelize.transaction` between `Drop.findOne` and `Drop.create`

Reply:
What is the use of `User` table in this faucet btw, just for record?

Reply:
> Run `Drop.create` first, then try catch `cosmosClient.distribute`

but we don't have `txHash` yet before running `cosmosClient.distribute`?

Reply:
we need the lock though, maybe use "pending" then update it after done?

Reply:
Used transaction and `lock` config. See if it's OK to just create a data row once.
(Since if we create a pending row first and then update `txHash` later, the transaction will still roll it back if tx is failed?)

ref: 
https://sequelize.org/docs/v6/other-topics/transactions/#locks
https://stackoverflow.com/a/70889313/7978205